### ISSUE TITLE
Improve toolbar UX

### ISSUE THREAD

**Issue by nadr0** [KittyCAD/modeling-app#4469]
When implementing the Center Rectangle tool I came across several UX issues within the toolbar. To prevent bloating center rectangle I've outlined these preexisting issues below. 

1. EscL or L, when active it shows Esc<Key> (keybinds), the copy is confusing.
2. If action button is focused and you press keybinds the tooltip is kept focused (the old one is kept opened)
3. Action array dropdown still shows focused tooltip when dropdown is open
4. Select the dropdown actionbutton properly
5. Select the tooltip actionbutton properly
6. If you click to disable the tool you still see the tooltip due to focus
7. If you have a tool selected then focus is applied the hover tooltip is under the focused tooltip
8. Allow for multiple icon types within dropdown menus.

https://github.com/user-attachments/assets/f7b424bf-8e72-4996-aad2-d21884121142

I'd recommend removing the `focus` logic on tooltips and keep it to `hover`.

I think a lot of tools do not have tooltips stay open when focused, usually on hover. Excalidraw does have a little description but that is separate from the Tooltip. We both embed a description keybind, usage all into a single tooltip. 

![Image](https://github.com/user-attachments/assets/8af14239-0757-43d8-8408-708f81960f5a)

**Comment by franknoirot** [comment]
The focus logic is in a probably naïve attempt at accessibility, but I support whatever we need to do to make these feel better including removing it and/or splitting out the description as you suggested. One thing to consider checking is if we've been using `:focus` where we should have been using [`:focus-visible`](https://developer.mozilla.org/en-US/docs/Web/CSS/:focus-visible#focus_vs_focus-visible), which should not appear on click interactions, etc.

**Comment by Irev-Dev** [comment]
I'll add this one here
https://github.com/KittyCAD/modeling-app/issues/5457
I noticed problem with the drop downs as well, there's also this (which was a change with multi-profile which I think could use a re-think)

https://github.com/user-attachments/assets/57d28759-b999-4d8f-9d09-2b9fe6717aa3

There are enough issues piled up here they probably need a bit of master plan of how we want the toolbar to be, rather than fixing piecemeal.

**Comment by Irev-Dev** [comment]
Frank asked I fixed esc to unequip the three point arc in https://github.com/KittyCAD/modeling-app/pull/5722

I though the fix might be to add 
```ts
hotkey: (state) => 
            state.matches({ Sketch: 'Arc three point tool' })
              ? ['Esc', 'T']
              : 'T',
```
To the toolbar for that tool, but this doesn't work, and found that the same problem already exists for the center rect, So I think this effect all tools in the drop downs.

**Comment by Irev-Dev** [comment]
Gonna give this one to you as well @andrewvarga.

This one is somewhere between a bug and a feature because there's some figuring out to do here, there will be two parts

1: coming up with a plan for what the final behaviours should be (run it past us here)
2: executing on that plan

Places to start looking are `src/lib/toolbar.ts` and `src/Toolbar.tsx`

How does that sound.

**Comment by andrewvarga** [comment]
I agree with the thoughts here, I think just like in Figma the tooltip should only appear for hover, that would resolve most of the issues arising from having a focused vs selected tool, so I would move that to hover only.
Those unrelated smaller bugs seem straightforward too, I'm looking into them.

**Comment by franknoirot** [comment]
> [@Irev-Dev](https://github.com/Irev-Dev) I've added Esc hotkey for the 3 point circle but there is an error, maybe something I can look into (?) Added a WIP PR here: [#5841](https://github.com/KittyCAD/modeling-app/pull/5841)

[This comment](https://github.com/KittyCAD/modeling-app/pull/5722#issuecomment-2728674609) seems relevant to this discussion, at about 1:00 into the video

**Comment by Irev-Dev** [comment]
Please do look into the error, I'm guessing some how xState is able to get into a weird state. Happy to pair on it as well.

**Comment by franknoirot** [comment]
@andrewvarga nice catch with that error, just checking in, do you need help debugging XState or to pair on this? I'm available too in case you want a partner and @Irev-Dev is in nighttime
